File: Array/c++/Max Sum Contiguous Subarray.cpp
int Solution::maxSubArray(const vector<int> &A) {
    
    int max_arr = -1001; //  maximum of entire array
    int max_i = 0; // maximum till the ith index
    
    for(int i = 0; i<A.size() ; ++i)
    {
        max_i = max_i + A[i];
        if(max_i < A[i])
        {
            max_i = A[i];
        }
        if(max_arr < max_i)
        {
            max_arr = max_i;
        }
    }
    
    return max_arr;
}

File: Array/python/Max Sum Contiguous Subarray.py
class Solution:
    # @param A : tuple of integers
    # @return an integer
    def maxSubArray(self, A):
        max_arr = -1001 #max of the whole array
        max_i = 0 #max till the ith index
        
        for i in range(len(A)):
            max_i+=A[i]
            if(max_i<A[i]):
                max_i = A[i] #changing max to current ith value
            if(max_arr < max_i):
                max_arr=max_i #changing overall max of the array
        return max_arr